pkg/kubelet/kuberuntime/kuberuntime_manager.go

```diff
@@ -641,12 +647,25 @@ func (m *kubeGenericRuntimeManager) computePodActions(pod *v1.Pod, podStatus *ku
 //  2. Kill pod sandbox if necessary.
 //  3. Kill any containers that should not be running.
 //  4. Create sandbox if necessary.
-//  5. Create ephemeral containers.
-//  6. Create init containers.
-//  7. Create normal containers.
-func (m *kubeGenericRuntimeManager) SyncPod(pod *v1.Pod, podStatus *kubecontainer.PodStatus, pullSecrets []v1.Secret, backOff *flowcontrol.Backoff) (result kubecontainer.PodSyncResult) {
+func (m *kubeGenericRuntimeManager) SyncPodSandbox(pod *v1.Pod, podStatus *kubecontainer.PodStatus, pullSecrets []v1.Secret, backOff *flowcontrol.Backoff) (result kubecontainer.PodSyncResult, podIPs []string, podSandboxID string) {
 	// Step 1: Compute sandbox and container changes.
 	podContainerChanges := m.computePodActions(pod, podStatus)
+
+	m.containerChangesLock.Lock()
+	if m.podContainerChanges == nil {
+		m.podContainerChanges = make(map[kubetypes.UID]podActions)
```

Comment:
I would like to avoid storing podContainerChanges if at all possible.  Is there a reason we can't just recompute podContainerChanges in SyncPod below?

Reply:
I haven't profiled computePodActions to know its cost.
Current formation is mostly to keep the flow as the existing one.

Reply:
as long as it doesn't make any CRI calls, it should be fine to add.